source/git/identifier.go

```diff
@@ -32,9 +32,10 @@ func NewGitIdentifier(remoteURL string) (*GitIdentifier, error) {
 	}
 
 	repo := GitIdentifier{Remote: u.Remote}
-	if u.Fragment != nil {
-		repo.Ref = u.Fragment.Ref
-		repo.Subdir = u.Fragment.Subdir
+	if u.Opts != nil {
+		repo.Ref = u.Opts.Ref
+		repo.Checksum = u.Opts.Checksum
```

Comment:
This is the place where it is getting checksum from querystring in LLB while it really comes from `pb.AttrGitChecksum`

Reply:
Looks like there is still a wrong type in here after parsing as the checksum should only come from props (I don't think this is some backwards compatibility?).